rotkehlchen/externalapis/manual_current_price.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.globaldb import GlobalDBHandler
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.interfaces import CurrentPriceOracleInterface
+from rotkehlchen.types import Price
+from rotkehlchen.user_messages import MessagesAggregator
+
+
+class ManualCurrentOracle(CurrentPriceOracleInterface):
+
+    def __init__(self, msg_aggregator: MessagesAggregator) -> None:
+        super().__init__(oracle_name='manual current price oracle')
+        self.msg_aggregator = msg_aggregator
+
+    def rate_limited_in_last(self, seconds: Optional[int] = None) -> bool:
+        return False
+
+    def query_current_price(self, from_asset: Asset, to_asset: Asset) -> Price:
+        """Searches for a manually specified current price of the `from_asset`. If finds
+        the price, converts it to a price in `to_asset` and returns."""
+        manual_current_result = GlobalDBHandler().get_manual_current_price(
+            asset=from_asset,
+        )
+        if manual_current_result is None:
+            return Price(ZERO)
+        current_to_asset, current_price = manual_current_result
+        try:
+            current_to_asset_price = Inquirer().find_price(
+                from_asset=current_to_asset,
+                to_asset=to_asset,
+            )
+            return Price(current_price * current_to_asset_price)
+        except RecursionError:
+            self.msg_aggregator.add_warning(
+                f'Was not able to find price from {str(from_asset)} to {str(to_asset)} since your '
+                f'manual current prices form a loop. For now, other oracles will be used.',
+            )
+            return Price(ZERO)
```

Comment:
1. Succesful return at the end.
2. Add a comment explaining when this may happen during RecursionError. We discusses when but we will forget it.

```suggestion
        except RecursionError:
            self.msg_aggregator.add_warning(
                f'Was not able to find price from {str(from_asset)} to {str(to_asset)} since your '
                f'manual current prices form a loop. For now, other oracles will be used.',
            )
            return Price(ZERO)

        return Price(current_price * current_to_asset_price)
```